clientd3d/music.c

```diff
@@ -146,22 +153,60 @@ DWORD OpenMidiFile(const char *lpszMIDIFileName)
 
    sprintf(filename, "%s%s", current_dir, lpszMIDIFileName);
    debug(("music filename = %s\n", filename));
-
-   memset(&mciOpenParms, 0, sizeof(MCI_OPEN_PARMS));
-   mciOpenParms.lpstrDeviceType = "sequencer";
-   mciOpenParms.lpstrElementName = filename;
-   if (dwReturn = mciSendCommand(0, MCI_OPEN, MCI_OPEN_ELEMENT,
+   // Is it a background music or jala music file?
+   if (case_bg)
+   {
+      if (playing_music != 0)
+      {
+         mciSendCommand(midi_bg_music_element, MCI_CLOSE, 0, 0);
```

Comment:
I'm not sure about mp3s.  I only just learned how MCI works this last week.

Yes we need to get all the midis into the database though.  Testing this change means you need a ton of mana because only Jig and Spellbane songs seem to exist in my resource folder.  I've given my tester guy every node, 70 mysticism and a few mints and he can only just barely hold up Jig long enough to make sure the music loops correctly.